### PR TITLE
Hotfix for resolving ADC = 0 issues

### LINKED ISSUE #39: Divide by zeros in tensor fitting
There are currently two places in tensor fitting where divide-by-zero could possibly occur. This produces lots of warnings (unless disabled) and possibly inaccurate kurtosis parameters.

The first is in [FA calculation](https://github.com/m-ama/PyDesigner/blob/09aa88749aec1286e9afa886dcba0af0d09b94bf/designer/fitting/dwipi.py#L661-L663) of `dwipi.py`. In particular 

The second occurs in AKC calculation when ADC = 0, in the function `kurtosisCoeff` at [this line](https://github.com/m-ama/PyDesigner/blob/09aa88749aec1286e9afa886dcba0af0d09b94bf/designer/fitting/dwipi.py#L338)

Here's a log of terminal output from one such erroneous runs for reference
```

```/Users/sid/Downloads/PyDesigner_Test/In/preprocessed_dwi.nii
Image preprocessed_dwi.nii loaded successfully
IRLLS: Noise Estimation : 100%|█| 187039/187039 [00:08<00:00, 23170.22vox/s]
IRLLS: Outlier Detection: 100%|█| 187039/187039 [00:59<00:00, 3140.20vox/s]
Constrained Tensor Fit  :   5%| | 9873/187039 [00:02<00:38, 4633.91vox/s]/Users/sid/miniconda3/envs/deep/lib/python3.7/site-packages/joblib/externals/loky/process_executor.py:706: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.
  "timeout or by a memory leak.", UserWarning
Constrained Tensor Fit  : 100%|█| 187039/187039 [00:46<00:00, 4034.20vox/s]
Detected DKI
DTI params              : 100%|█| 187039/187039 [00:13<00:00, 13905.56vox/s]
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:663: RuntimeWarning: divide by zero encountered in true_divide
  l1 ** 2 + l2 ** 2 + l3 ** 2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:663: RuntimeWarning: invalid value encountered in true_divide
  l1 ** 2 + l2 ** 2 + l3 ** 2)
AKC Outlier Detection   : 100%|██████| 10/10 [00:23<00:00,  2.34s/blk]
AKC Correction          : 100%|███| 21/21 [00:10<00:00,  2.07tensor/s]
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   1%| | 2523/187039 [00:00<06:33, 468.83vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   2%| | 3491/187039 [00:00<04:40, 654.43vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   3%| | 5219/187039 [00:01<03:17, 918.52vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   4%| | 7811/187039 [00:01<01:44, 1713.52vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   5%| | 9780/187039 [00:01<01:19, 2219.63vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   7%| | 12707/187039 [00:02<00:57, 3016.26vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :   8%| | 15615/187039 [00:02<00:34, 4934.08vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  18%|▏| 32743/187039 [00:04<00:17, 8623.43vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  18%|▏| 33875/187039 [00:04<00:17, 8741.16vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  26%|▎| 48995/187039 [00:06<00:16, 8456.37vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  30%|▎| 56771/187039 [00:07<00:16, 7683.50vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  32%|▎| 59651/187039 [00:07<00:15, 7990.62vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  40%|▍| 75635/187039 [00:09<00:12, 8624.48vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  44%|▍| 82118/187039 [00:10<00:12, 8479.08vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  45%|▍| 84995/187039 [00:10<00:12, 8306.00vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  48%|▍| 89747/187039 [00:11<00:11, 8500.36vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  48%|▍| 90611/187039 [00:11<00:11, 8490.68vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              :  57%|▌| 106739/187039 [00:13<00:09, 8541.98vox/s]/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: invalid value encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
/Users/sid/Repos/PyDesigner/designer/fitting/dwipi.py:338: RuntimeWarning: divide by zero encountered in true_divide
  akc = (akc * np.tile(md**2, (adc.shape[0], 1)))/(adc**2)
DKI params              : 100%|█| 187039/187039 [00:22<00:00, 8151.34vox/s]

### PR DESCRIPTION
**Pull request by TheJaeger**
Closes #39

Defines a precision of `1E-8` when diffusion coefficient is zero to prevent `RuntimeWarning: invalid value encountered in true_divide` warnings. This also produces more accurate kurtosis parameters

### CODE REVIEW

**Comment by TheJaeger** [comment]
Squash and merge if it looks good.

File: designer/fitting/dwipi.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from joblib import Parallel, delayed
 from tqdm import tqdm
 import random as rnd
-import warnings
+# import warnings
```
Comment (jbteves) [inline designer/fitting/dwipi.py]: Please don't comment this, remove it.

File: designer/fitting/dwipi.py [hunk 2]
```diff
@@ -331,6 +331,7 @@ def kurtosisCoeff(self, dt, dir):
         wcnt, wind = self.createTensorOrder(4)
         ndir = dir.shape[0]
         adc = self.diffusionCoeff(dt[:6], dir)
+        adc[adc == 0] = minZero
```
Comment (jbteves) [inline designer/fitting/dwipi.py]: That's perfect, this part LGTM

File: designer/fitting/dwipi.py (hunk 1 above)
Comment (TheJaeger) [reply designer/fitting/dwipi.py]: Changes made in the other pull request.

**Comment by jbteves** [comment]
I think we should close this one.

**Comment by TheJaeger** [comment]
> I think we should close this one.

Yeah, let me close this particular request and submit a new one that doesn't conflict with the other pull request.

**Pull request closed without merging**